build-support/bin/release.sh

```diff
@@ -7,9 +7,22 @@ set -e
 ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && cd "$(git rev-parse --show-toplevel)" && pwd)
 source ${ROOT}/build-support/common.sh
 
-PY=$(which python2.7 || exit 0)
-[[ -n "${PY}" ]] || die "You must have python2.7 installed and on the path to release."
-export PY
+# Set the Python interpreter to be used for the virtualenv. Note we allow the user to
+# predefine this value so that they may point to a specific interpreter, e.g. 2.7.13 vs. 2.7.15.
+export PY="${PY:-python2.7}"
+if ! which "${PY}" >/dev/null; then
+  die "Python interpreter ${PY} not discoverable on your PATH."
+fi
+py_major_minor=$(${PY} -c 'import sys; print(".".join(map(str, sys.version_info[0:2])))')
+if [[ "${py_major_minor}" != "2.7" ]]; then
+  die "Invalid interpreter. The release script requires python2.7, and you are using python${py_major_minor}."
+fi
```

Comment:
@benjyw I split out the original one check into two checks to verify 1) the interpreter is on the PATH and 2) it's specifically python2.7, as we don't support Py3 for release.sh (yet).